notification.go

```diff
@@ -0,0 +1,107 @@
+package ts3
+
+import (
+	"errors"
+	"strings"
+)
+
+// Notify event types
+// Notifications are very badly documented by TeamSpeak;
+// A complete but unofficial documentation in German can be found here:
+// http://yat.qa/ressourcen/server-query-notify/
+const (
+	// ServerEvents registers the following events:
+	// `cliententerview`, `clientleftview`, `serveredited`
+	ServerEvents = "server"
+
+	// ServerEvents registers the following events:
+	// `cliententerview`, `clientleftview`, `channeldescriptionchanged`, `channelpasswordchanged`
+	// `channelmoved`, `channeledited`, `channelcreated`, `channeldeleted`, `clientmoved`
+	ChannelEvents = "channel"
+
+	// TextServerEvents registers the `textmessage` event with `targetmode = 3`
+	TextServerEvents = "textserver"
+
+	// TextChannelEvents registers the `textmessage` event with `targetmode = 2`
+	//
+	// Notifications are only received for messages that are sent in the channel that the client is in.
+	TextChannelEvents = "textchannel"
+
+	// TextPrivateEvents registers the `textmessage` event with `targetmode = 1`
+	TextPrivateEvents = "textprivate"
+
+	// TokenUsedEvents registers the `tokenused` event
+	TokenUsedEvents = "tokenused"
+)
+
+// Notification contains the information of a notify event
+type Notification struct {
+	Type string
+	Data map[string]string
+}
+
+// SetNotifyHandler registers a func that handles received notifications
+func (c *Client) SetNotifyHandler(event string, handler func(Notification)) {
+	c.notifyHandler = handler
+}
+
+// NotifyRegister unregisters a given event
+//
+// The id can only be set for channel events
+// It's not possible to subscribe to multiple channel id's;
+// To receive events for all channels the id can be left out or set to 0.
+func (c *Client) NotifyRegister(event string, id ...uint) error {
```

Comment:
using a var arg here is a little ambiguous given the additional criteria as well it would be more user friendly to split it out to a couple of methods, calling a single internal one.
Also using a specific type with predefined constants for event would give stronger validation.